core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -114,21 +164,28 @@ thread_spawn_wrapper(wasm_exec_env_t exec_env, uint32 start_arg)
     }
 
     thread_start_arg->thread_id = thread_id = allocate_thread_id();
+    if (thread_id < 0) {
+        LOG_ERROR("Failed to get thread identifier");
+        goto thread_spawn_fail;
+    }
     thread_start_arg->arg = start_arg;
     thread_start_arg->start_func = start_func;
 
     os_mutex_lock(&exec_env->wait_lock);
     ret = wasm_cluster_create_thread(exec_env, new_module_inst, thread_start,
                                      thread_start_arg);
     if (ret != 0) {
-        os_mutex_unlock(&exec_env->wait_lock);
         LOG_ERROR("Failed to spawn a new thread");
-        goto thread_spawn_fail;
+        goto tid_allocation_fail;
     }
     os_mutex_unlock(&exec_env->wait_lock);
 
     return thread_id;
 
+tid_allocation_fail:
```

Comment:
i feel this label is a bit confusing because it's actually used when tid allocation succeeded.